services/alert/handlers.go

```diff
@@ -341,30 +347,31 @@ func (h *publishHandler) Handle(event alert.Event) {
 	}
 }
 
-type StateChangesOnlyHandlerConfig struct {
-	topics *alert.Topics
-}
-
-type stateChangesOnlyHandler struct {
-	topics *alert.Topics
-	logger *log.Logger
-	next   alert.Handler
-}
-
-func NewStateChangesOnlyHandler(c StateChangesOnlyHandlerConfig, l *log.Logger) handlerAction {
-	return &stateChangesOnlyHandler{
-		topics: c.topics,
-		logger: l,
-	}
-}
-
-func (h *stateChangesOnlyHandler) Handle(event alert.Event) {
-	if event.State.Level != event.PreviousState().Level {
-		h.next.Handle(event)
-	}
-}
-
-func (h *stateChangesOnlyHandler) SetNext(n alert.Handler) {
-	h.next = n
-}
-func (h *stateChangesOnlyHandler) Close() {}
+// TODO implement state changes only as a match condition
```

Comment:
Ah, yes so with the change to not have handler `actions` anymore there is no way to do stateChangesOnly via a handler.

The proposed solution is to use match criteria on handlers. So something like this

```yaml
id: my_handler
topic: system
kind: slack
   options:
       channel: '#alerts'
match: "changed" == TRUE AND "sometag" == 'bob'
```

Where the `match` property is a lambda expression then when it evals to true then the handler handles the alert event. This way we get the same functionality back but in a more consistent manner.

I was thinking we should add the `match` expression logic in another PR so that its can remain as an independent change. Or should we lump it in here so that we maintain feature parity from the get go? Thoughts?